src/openrct2/drawing/Drawing.String.cpp

```diff
@@ -1075,45 +1075,28 @@ void GfxDrawStringWithYOffsets(
     dpi.lastStringPos = { info.x, info.y };
 }
 
-void ShortenPath(utf8* buffer, size_t bufferSize, const utf8* path, int32_t availableWidth, FontStyle fontStyle)
+u8string ShortenPath(const u8string& path, int32_t availableWidth, FontStyle fontStyle)
 {
-    size_t length = strlen(path);
-
-    // Return full string if it fits
-    if (GfxGetStringWidth(const_cast<char*>(path), fontStyle) <= availableWidth)
+    if (GfxGetStringWidth(path, fontStyle) <= availableWidth)
     {
-        SafeStrCpy(buffer, path, bufferSize);
-        return;
+        return path;
     }
 
-    // Count path separators
-    int32_t path_separators = 0;
-    for (size_t x = 0; x < length; x++)
-    {
-        if (path[x] == *PATH_SEPARATOR || path[x] == '/')
-        {
-            path_separators++;
-        }
-    }
-
-    // TODO: Replace with unicode ellipsis when supported
-    SafeStrCpy(buffer, "...", bufferSize);
+    u8string shortenedPath = u8"...";
 
-    // Abbreviate beginning with xth separator
-    int32_t begin = -1;
-    for (int32_t x = 0; x < path_separators; x++)
+    size_t begin = 0;
+    while (begin < path.size())
     {
-        do
-        {
-            begin++;
-        } while (path[begin] != *PATH_SEPARATOR && path[begin] != '/');
+        begin = path.find_first_of(*PATH_SEPARATOR, begin + 1);
```

Comment:
Is the path separator always guaranteed to be just 1 char wide? Otherwise, you might want to `+ strlen(PATH_SEPARATER)` instead.

Reply:
I don't think any OS has anything else then / or \, I might be mistaken but thats what I know.

Reply:
```suggestion
        begin = path.find_first_of(*PATH_SEPARATOR, begin + strlen(PATH_SEPARATER));
```
For our glorious OSes with weird paths!

Or just ignore this and merge as is.